src/M6Web/Bundle/DaemonBundle/Command/DaemonCommand.php

```diff
@@ -2,114 +2,93 @@
+use Psr\EventDispatcher\EventDispatcherInterface;
 use React\EventLoop\LoopInterface;
-use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
 use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
-use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 
 /**
  * Class DaemonCommand
  * Abstract class for build daemon commands.
  */
-abstract class DaemonCommand extends ContainerAwareCommand
+abstract class DaemonCommand extends Command
 {
-    /**
-     * Tells if shutdown is requested.
-     *
-     * @var bool
-     */
+    /** @var bool tells if shutdown is requested */
     protected $shutdownRequested = false;
 
-    /**
-     * Alows the concrete command to
-     * setup an exit code.
-     *
-     * @var int
-     */
+    /** @var int allows the concrete command to setup an exit code */
     protected $returnCode = 0;
 
-    /**
-     * Loop count.
-     *
-     * @var int
-     */
+    /** @var int loop count */
     protected $loopCount = 0;
 
-    /**
-     * Store max loop option value.
-     *
-     * @var int
-     */
-    protected $loopMax = null;
+    /** @var int store max loop option value */
+    protected $loopMax;
 
-    /**
-     * Store max memory option value.
-     *
-     * @var int
-     */
+    /** @var int store max memory option value */
     protected $memoryMax = 0;
 
-    /**
-     * Store shutdown on exception option value.
-     *
-     * @var bool
-     */
+    /** @var bool store shutdown on exception option value */
     protected $shutdownOnException;
 
-    /**
-     * Display or not exception on command output.
-     *
-     * @var bool
-     */
+    /** @var bool display or not exception on command output */
     protected $showExceptions;
 
-    /**
-     * @var EventDispatcherInterface
-     */
-    protected $dispatcher = null;
+    /** @var ?EventDispatcherInterface */
+    protected $dispatcher;
 
-    /**
-     * @var LoopInterface
-     */
+    /** @var LoopInterface */
     protected $loop;
 
-    /**
-     * @var callable
-     */
+    /** @var callable */
     protected $loopCallback;
 
-    /**
-     * @var \Exception
-     */
-    protected $lastException = null;
+    /** @var \Exception */
+    protected $lastException;
 
-    /**
-     * @var float
-     */
-    protected $startTime = null;
+    /** @var float */
+    protected $startTime;
 
-    /**
-     * In seconds
-     *
-     * @var float
-     */
+    /** @var float time in seconds */
     protected $nextIterationSleepingTime = 0.0;
 
-    /**
-     * @var array
-     */
+    /** @var array */
     protected $iterationsEvents = [];
 
-    /**
-     * @var array
-     */
+    /** @var array */
     protected $iterationsIntervalCallbacks = [];
 
+    public function setEventDispatcher(EventDispatcherInterface $dispatcher = null): DaemonCommand
```

Comment:
Couldn't this be autowired and go into the construct ? valid also for `setLoop` and `setIterationsEvents`